src/renderer/decoder/common.js

```diff
@@ -0,0 +1,123 @@
+import { getJoinNormal, getLineNormal, neg } from '../../utils/geometry';
+
+/**
+ * Create a triangulated lineString: zero-sized, vertex-shader expanded triangle list
+ * with `miter` joins. For angle < 60 joins are automatically adjusted to `bevel`.
+ */
+export function addLineString (lineString, geomBuffer, index, isPolygon, skipCallback) {
+    let prevPoint, currentPoint, nextPoint;
+    let prevNormal, nextNormal;
+    let drawLine;
+
+    // We need at least two points
+    if (lineString.length >= 4) {
+        // Initialize the first two points
+        prevPoint = [lineString[0], lineString[1]];
+        currentPoint = [lineString[2], lineString[3]];
+        prevNormal = getLineNormal(prevPoint, currentPoint);
+
+        for (let i = 4; i <= lineString.length; i += 2) {
+            drawLine = !(skipCallback && skipCallback(i));
+
+            if (drawLine) {
```

Comment:
It would be nice to document how each triangle is positioned... Could we put a link to an image?

Reply:
You can find more information here: https://github.com/CartoDB/carto-vl/pull/219. I could add the information in the code.